superset/viz.py

```diff
@@ -104,6 +104,46 @@
 ]
 
 
+import re
+
+from geopy.format import DEGREE, DOUBLE_PRIME, PRIME
+
+LNG_LAT_POINT_PATTERN = re.compile(
+    r"""
+    .*?
+    (?P<longitude>
+      (?P<longitude_direction_front>[EW])?[ ]*
+      (?P<longitude_degrees>[+-]?{FLOAT})(?:[{DEGREE}D\*\u00B0\s][ ]*
+      (?:(?P<longitude_arcminutes>{FLOAT})[{PRIME}'m][ ]*)?
+      (?:(?P<longitude_arcseconds>{FLOAT})[{DOUBLE_PRIME}"s][ ]*)?
+      )?(?P<longitude_direction_back>[EW])?)
+    {SEP}
+    (?P<latitude>
+      (?P<latitude_direction_front>[NS])?[ ]*
+        (?P<latitude_degrees>[+-]?{FLOAT})(?:[{DEGREE}D\*\u00B0\s][ ]*
+        (?:(?P<latitude_arcminutes>{FLOAT})[{PRIME}'m][ ]*)?
+        (?:(?P<latitude_arcseconds>{FLOAT})[{DOUBLE_PRIME}"s][ ]*)?
+        )?(?P<latitude_direction_back>[NS])?)(?:
+    {SEP}
+      (?P<altitude>
+        (?P<altitude_distance>[+-]?{FLOAT})[ ]*
+        (?P<altitude_units>km|m|mi|ft|nm|nmi)))?
+    \s*$
+""".format(
+        FLOAT=r"\d+(?:\.\d+)?",
+        DEGREE=DEGREE,
+        PRIME=PRIME,
+        DOUBLE_PRIME=DOUBLE_PRIME,
+        SEP=r"\s*[,;/\s]\s*",
+    ),
+    re.VERBOSE | re.UNICODE,
```

Comment:
If you're the author of this regex, could you add a comment above it with an explanation (since it's quite complex)? Otherwise, could you add a comment with the source?

Reply:
I refer to the `geopy.point` implementation of this regex.

https://github.com/geopy/geopy/blob/f495974c32a7a7b1eb433e7b8c87166e96375c32/geopy/point.py#L14-L40

-------

This expression can be understood based on the group name.

The whole is in three group:

1. match longitude
2. match latitude
3. match altitude

longitude matcher is similar as latitude.

match degrees, arcminutes, arcseconds, direction_front